qiskit_experiments/calibration/calibrations.py

```diff
@@ -0,0 +1,798 @@
+            valid_only: Use only parameters marked as valid.
+            group: The calibration group from which to draw the parameters.
+                If not specified this defaults to the 'default' group.
+            cutoff_date: Retrieve the most recent parameter up until the cutoff date. Parameters
+                generated after the cutoff date will be ignored. If the cutoff_date is None then
+                all parameters are considered. This allows users to discard more recent values that
+                may be erroneous.
+
+        Returns:
+            value: The value of the parameter.
+
+        Raises:
+            CalibrationError:
+                - If there is no parameter value for the given parameter name and pulse channel.
+        """
+
+        # 1) Identify the parameter object.
+        param_name = param.name if isinstance(param, Parameter) else param
+        sched_name = schedule.name if isinstance(schedule, (Schedule, ScheduleBlock)) else schedule
+
+        param = self.calibration_parameter(param_name, qubits, sched_name)
+
+        # 2) Get a list of candidate keys restricted to the qubits of interest.
+        candidate_keys = []
+        for key in self._parameter_map_r[param]:
+            candidate_keys.append(ParameterKey(key.schedule, key.parameter, qubits))
+
+        # 3) Loop though the candidate keys to candidate values
+        candidates = []
+        for key in candidate_keys:
+            if key in self._params:
+                candidates += self._params[key]
+
+        # If no candidate parameter values were found look for default parameters
+        # i.e. parameters that do not specify a qubit.
+        if len(candidates) == 0:
+            for key in candidate_keys:
+                if ParameterKey(key.schedule, key.parameter, None) in self._params:
+                    candidates += self._params[ParameterKey(key.schedule, key.parameter, None)]
+
+        # 4) Filter candidate parameter values.
+        if valid_only:
+            candidates = [val for val in candidates if val.valid]
+
+        candidates = [val for val in candidates if val.group == group]
+
+        if cutoff_date:
+            candidates = [val for val in candidates if val.date_time <= cutoff_date]
+
+        if len(candidates) == 0:
+            msg = f"No candidate parameter values for {param_name} in calibration group {group} "
+
+            if qubits:
+                msg += f"on qubits {qubits} "
+
+            if sched_name:
+                msg += f"in schedule {sched_name} "
+
+            if cutoff_date:
+                msg += f"with cutoff date: {cutoff_date}"
+
+            raise CalibrationError(msg)
+
+        # 5) Return the most recent parameter.
+        return max(candidates, key=lambda x: x.date_time).value
+
+    def get_schedule(
+        self,
+        name: str,
+        qubits: Tuple[int, ...],
+        free_params: List[Tuple[str, str, Tuple]] = None,
```

Comment:
It is nice to support free parameters within subschedules. What do you think about allowing simple strings for `free_params` the way it was before as well. A string like `"amp"` in `free_params` would be an alias for `(name, "amp", qubits)` where `name` and `qubits` are the arguments to `get_schedule`.

Reply:
Sounds good. See here: 980f2c2